### PR TITLE
Add unit tests for Controllers and DALs

### PR DESCRIPTION
**Pull request by oxyflush**

### CODE REVIEW

File: src/test/java/com/fryrank/controller/MessageControllerTests.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import static com.fryrank.Constants.WELCOME_MESSAGE;
+
+public class MessageControllerTests {
+
+    @Autowired
+    private MessageController controller;
+
+    @Test
+    public void testGetWelcomeMessage() throws Exception {
+        MessageController controller = new MessageController();
```
Comment (NickPriv) [inline src/test/java/com/fryrank/controller/MessageControllerTests.java:17]: Remove this

File: src/test/java/com/fryrank/controller/MessageControllerTests.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+        String actualMessage = controller.getWelcomeMessage();
+        String expectedMessage = WELCOME_MESSAGE;
+
+        Assert.assertEquals(actualMessage, expectedMessage);
```
Comment (NickPriv) [inline src/test/java/com/fryrank/controller/MessageControllerTests.java:22]: Would recommend importing assertEquals only instead of importing entire Assert class

File: src/test/java/com/fryrank/controller/ReviewControllerTests.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    public void testAddNewReviewForRestaurant() throws Exception {
+        Review expectedReview = new Review("review_id_1", "restaurant_id_1", "author_id_1", 5.0 , "title_1", "body_1");
+
+        Mockito.when(reviewDAL.addNewReview(expectedReview)).thenReturn(expectedReview);
+
+        Review actualReview = controller.addNewReviewForRestaurant(expectedReview);
+
+        Assert.assertEquals(expectedReview, actualReview);
+    }
+
```
Comment (NickPriv) [inline src/test/java/com/fryrank/controller/ReviewControllerTests.java:50]: Can you add tests for null input cases? For example, restaurantId is null, review is null, and some attribute of the Review object is null

File: src/test/java/com/fryrank/dal/ReviewDALTests.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    public void testGetAllReviewsByRestaurantId() throws Exception {
+        Query query = new Query();
+        query.addCriteria(Criteria.where("restaurantId").is("1"));
+        List<Review> expectedReviews = new ArrayList<>() {
+            {
+                add(new Review("review_id_1", "restaurant_id_1", "author_id_1", 5.0 , "title_1", "body_1"));
+                add(new Review("review_id_2", "restaurant_id_2", "author_id_2", 7.0 , "title_2", "body_2"));
+            }
+        };
+        Mockito.when(mongoTemplate.find(query, Review.class)).thenReturn(expectedReviews);
```
Comment (NickPriv) [inline src/test/java/com/fryrank/dal/ReviewDALTests.java:36]: Recommend importing when method specifically instead of entire Mockito

File: src/test/java/com/fryrank/dal/ReviewDALTests.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Test
+    public void testAddNewReview() throws Exception {
+        Review expectedReview = new Review("review_id_1", "restaurant_id_1", "author_id_1", 5.0 , "title_1", "body_1");
+
+        Review actualReview = reviewDAL.addNewReview(expectedReview);
+
+        Assert.assertEquals(expectedReview, actualReview);
+    }
+}
```
Comment (NickPriv) [inline src/test/java/com/fryrank/dal/ReviewDALTests.java:51]: Can you add test for no review returned from DB for a given restaurantId?

**Review by NickPriv — CHANGES_REQUESTED** [review]

File: src/test/java/com/fryrank/dal/ReviewDALTests.java (hunk 5 above)
Comment (oxyflush) [reply src/test/java/com/fryrank/dal/ReviewDALTests.java:51]: I think in these unit tests we're more testing the code path execution than the actual outputs since we're mocking the calls to the DB. Where the output would matter is in the web layer where we should make sure to return an empty JSON, which would probably be better done in an integration test.

File: src/test/java/com/fryrank/controller/ReviewControllerTests.java (hunk 3 above)
Comment (oxyflush) [reply src/test/java/com/fryrank/controller/ReviewControllerTests.java:50]: I think this is a similar case to https://github.com/NickPriv/FryRankBackend/pull/47/files#r1500227397, where we're really only testing the code path.

Comment (NickPriv) [reply src/test/java/com/fryrank/controller/ReviewControllerTests.java:51]: I think it's different because we have @NonNull annotations which technically have two conditions that can be tested, null or non null parameters being provided.

Comment (oxyflush) [reply src/test/java/com/fryrank/controller/ReviewControllerTests.java:51]: Ah testing the annotation makes sense.

My above comment was more directed to restaurant ID and other attributes are null part.

File: src/test/java/com/fryrank/controller/MessageControllerTests.java (hunk 2 above)
Comment (oxyflush) [reply src/test/java/com/fryrank/controller/MessageControllerTests.java:18]: Done in e560385f84676608e8101263f9702b6e5c9c5865.

File: src/test/java/com/fryrank/dal/ReviewDALTests.java (hunk 4 above)
Comment (oxyflush) [reply src/test/java/com/fryrank/dal/ReviewDALTests.java]: Done in b1cd8234a3db8d047ba0a0127860fc45e407be8a

File: src/test/java/com/fryrank/controller/MessageControllerTests.java (hunk 1 above)
Comment (oxyflush) [reply src/test/java/com/fryrank/controller/MessageControllerTests.java]: Done in b1cd8234a3db8d047ba0a0127860fc45e407be8a

File: src/test/java/com/fryrank/controller/ReviewControllerTests.java (hunk 3 above)
Comment (oxyflush) [reply src/test/java/com/fryrank/controller/ReviewControllerTests.java:56]: Done in d068ca1e463c3737dbe88c9adc9cfac2249427e6.

Comment (oxyflush) [reply src/test/java/com/fryrank/controller/ReviewControllerTests.java:84]: Done in 6cbdb1d3f16e40ee7a0515c1bb07f5b4ee96e6c9.

**Review by NickPriv — APPROVED** [review]
